### PR TITLE
fix: proxy settings do not take effect when switched to system or disabled

### LINKED ISSUE #9625: Proxy settings do not take effect when switched to system or disabled
### Bug description

Proxy settings do not take effect when switched to system or disabled. After restart, only the first setting is valid.

https://github.com/user-attachments/assets/2d74256e-15ec-4ee5-a41d-e8262f4492eb

### Operating system

Windows 11

### Installation Method

Installer from website/GitHub releases

### Version

1.13.3

### Steps to reproduce

1. Set system proxy setting to `127.0.0.1:7890`
2. Select `System` in Podman Desktop -> Setting -> Proxy
3. Restart podman desktop
4. Push `Update` in Podman Desktop -> Setting -> Proxy. This will set env `http_proxy=http:127.0.0.1:7890` in the config file.
5. Select Proxy configuration to `Manual`. Change Web Proxy (HTTP) to `http:127.0.0.1:7891` and push `Update`. This will set env `http_proxy=http:127.0.0.1:7891` in the config file.
6. Select Proxy configuration to `System` and push `Update`. This did not change the config file.
7. Select Proxy configuration to `Disabled` and push `Update`. This did not change the config file.

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #9625

### What does this PR do?

Ensure events are correctly sent when switching proxy state

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#9625 

### How to test this PR?

Update proxy settings

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/proxy.ts [hunk 1]
```diff
@@ -178,15 +179,16 @@ export class Proxy {
...
 
     // update configuration
     const proxyConfiguration = this.configurationRegistry.getConfiguration('proxy');
-    proxyConfiguration.update('enabled', state).catch((err: unknown) => {
-      console.error('Error updating proxy state', err);
-    });
+    await proxyConfiguration.update('enabled', state);
 
+    if (state === ProxyState.PROXY_SYSTEM) {
+      await this.setProxy({} as ProxySettings);
```
Comment (benoitf) [inline packages/main/src/plugin/proxy.ts]: I think setProxy should accept undefined as a parameter and here we call setProxy(undefined)

**Review by benoitf — APPROVED** [review]
issue(non-blocking): my POV is that we should not assign new values to the parameters of a function (immutable)
probably another rule to enforce in the linter

code is OK

nitpick: the title of the PR needs to be fixed (it ends with 'dis...')

**Comment by benoitf** [comment]
the rule is https://eslint.org/docs/latest/rules/no-param-reassign

**Review by benoitf — APPROVED** [review]
thx for the changes

need to fix the title of the PR before merging

**Comment by benoitf** [comment]
@jeffmaury do we need to backport the fix to 1.14.x branch ?

**Comment by jeffmaury** [comment]
I would not qualifier as a blocker as it's a specific use case but would be a nice plus

**Comment by benoitf** [comment]
@Mergifyio backport 1.14.x

**Pull request merged**